stdlib/Serialization/src/Serialization.jl

```diff
@@ -22,7 +22,8 @@ mutable struct Serializer{I<:IO} <: AbstractSerializer
     table::IdDict{Any,Any}
     pending_refs::Vector{Int}
     known_object_data::Dict{UInt64,Any}
-    Serializer{I}(io::I) where I<:IO = new(io, 0, IdDict(), Int[], Dict{UInt64,Any}())
+    version::Int
```

Comment:
This is probably a good idea, but I still usually try to handle these by making the data fully self-describing, i.e. store the new field in such a way that the type of value you deserialize tells you which version of the object you have.

Reply:
That's what I tried at first, but it just becomes super messy, for what seems to be little benefit. I don't see a realistic way to support non-versioned serialization blobs stably, so we might as well require the version.